waitress/parser.py

```diff
@@ -291,8 +291,12 @@ def crack_first_line(line):
             version = m.group(5)
         else:
             version = None
-        command = m.group(1).upper()
+        method = m.group(1)
+
+        # the
+        if method != method.upper():
+            raise ParsingError('Malformed HTTP method "%s"' % tostr(method))
```

Comment:
Yeah, nginx and other servers drop the incoming request on the floor. This caused me a lot of headache and debugging issues when I was testing with waitress locally and using nginx in production.

All currently defined HTTP methods are uppercase, I don't think it is a bad idea to enforce that.